### PR TITLE
[Japneet | Procheta] Add. component test cases

### PR DESCRIPTION
**Pull request by japneetbhatia**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -9,6 +10,7 @@ const App = () => {
         <Routes>
           <Route path="/" element={<SearchScreen />} />
           <Route path="/search" element={<SearchResult />} />
+          <Route path="/invoice" element={<InvoiceScreen />} />
```
Comment (hariharanweb) [inline src/App.js:13]: Should We use the same name as in schema; `Quote`? verify with schema

File: src/components/Catalog.js [hunk 2]
```diff
@@ -20,15 +20,17 @@ const bppProvider = (provider) => {
...
       {provider.items.map((item) => (
-        <Item key={item.id} item={item} />
+        <Item key={item.id} item={item} details={details} />
       ))}
     </Grid>
   );
 };
 
-const Catalog = ({ catalog }) => {
+const Catalog = ({ catalog, ...props }) => {
```
Comment (hariharanweb) [inline src/components/Catalog.js:29]: lets avoid using ...props. Instead use explicit values

File: src/components/Catalog.spec.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    render(<Catalog catalog={catalog} />);
+    const displayedImage = document.querySelector("img");
+    expect(displayedImage.src).toContain(
+      "https://cdn3.iconfinder.com/data/icons/fake-news/500/yul748_24_fake_news_truck_business_logo_computer_car-512.png"
+    );
+  });
+
+  it("should display catalog item", async () => {
+    const { getByTestId } = render(<Catalog catalog={catalog} />);
+    expect(getByTestId(/Item/)).toBeInTheDocument();
```
Comment (hariharanweb) [inline src/components/Catalog.spec.js:106]: Can we avoid using regex

File: src/components/Invoice.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        Pick-up Location: {fromLocation}
+      </Typography>
+      <Typography variant="body1" display="block" gutterBottom>
+        Destination Location: {toLocation}
+      </Typography>
+    </Box>
+  </>
+);
+
+export default Invoice;
```
Comment (hariharanweb) [inline src/components/Invoice.js:84]: Lets use Quote if same as core schema?

File: src/components/Journey.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  left: "50%",
+  transform: "translate(-50%, -50%)",
+  width: 400,
+  bgcolor: "background.paper",
+  border: "2px solid #000",
+  boxShadow: 24,
+  p: 4,
+};
+
+const Journey = ({ bookingInformation, bookingResponse }) => {
```
Comment (hariharanweb) [inline src/components/Journey.js:18]: Lets try to use same item as in core schema

File: src/components/Loader.spec.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import React from "react";
+import { render, waitFor, screen } from "@testing-library/react";
+import "@testing-library/jest-dom";
+import Loader from "./Loader";
+
+jest.mock("react-router-dom", () => {
```
Comment (hariharanweb) [inline src/components/Loader.spec.js:6]: why are we doing this mock

File: src/App.js (hunk 1 above)
Comment (japneetbhatia) [reply src/App.js:13]: The Quote is selectJourney Modal and this invoice is final confirmation screen.
Though we specify route for invoice the invoice is not integrated in workflow till now.

File: src/components/Catalog.js (hunk 2 above)
Comment (prochetab1999) [reply src/components/Catalog.js]: updated

File: src/components/Catalog.spec.js (hunk 3 above)
Comment (prochetab1999) [reply src/components/Catalog.spec.js:106]: Updated

File: src/components/Invoice.js (hunk 4 above)
Comment (prochetab1999) [reply src/components/Invoice.js:84]: The Quote is selectJourney Modal and this invoice is final confirmation screen.
Though we specify route and also component for invoice, by mistake, the invoice is not integrated in workflow till now.

File: src/components/Loader.spec.js (hunk 6 above)
Comment (prochetab1999) [reply src/components/Loader.spec.js:6]: It was needed in screens, but not needed in components. So removed it.

File: src/components/Journey.js (hunk 5 above)
Comment (prochetab1999) [reply src/components/Journey.js]: Updated

**Pull request merged**